In [2]:
import pandas as pd

In [38]:
hospitals = pd.read_csv('../data/Hospitals.csv')

In [4]:
hospitals.head(2)

X             Y  OBJECTID        ID  \
0 -1.331889e+07  4.346975e+06         1   5793230   
1 -1.322651e+07  4.049626e+06         2  53391362   

                                                NAME                  ADDRESS  \
0                    CENTRAL VALLEY GENERAL HOSPITAL  1025 NORTH DOUTY STREET   
1  LOS ROBLES HOSPITAL & MEDICAL CENTER - EAST CA...           150 VIA MERIDA   

              CITY STATE    ZIP           ZIP4  ...             VAL_DATE  \
0          HANFORD    CA  93230  NOT AVAILABLE  ...  2014/02/10 00:00:00   
1  WESTLAKE VILAGE    CA  91362  NOT AVAILABLE  ...  2014/02/10 00:00:00   

                            WEBSITE       STATE_ID       ALT_NAME ST_FIPS  \
0      http://www.hanfordhealth.com  NOT AVAILABLE  NOT AVAILABLE       6   
1  http://www.losrobleshospital.com  NOT AVAILABLE  NOT AVAILABLE       6   

         OWNER TTL_STAFF  BEDS         TRAUMA  HELIPAD  
0  PROPRIETARY      -999    49  NOT AVAILABLE        N  
1  PROPRIETARY      -999    62  NOT AVAILABLE        N  

[2 rows x 34 columns]

In [5]:
hospitals.tail(2)

X             Y  OBJECTID         ID  \
7594 -8.522423e+06  4.718512e+06      7610  191021401   
7595 -1.244921e+07  4.976468e+06      7611  196784108   

                                    NAME                     ADDRESS  \
7594  J KENT MCNEW FAMILY MEDICAL CENTER  175 HARRY S TRUMAN PARKWAY   
7595       UNIVERSITY ORTHOPAEDIC CENTER              590 WAKARA WAY   

                CITY STATE    ZIP           ZIP4  ...             VAL_DATE  \
7594       ANNAPOLIS    MD  21401  NOT AVAILABLE  ...  2020/05/05 00:00:00   
7595  SALT LAKE CITY    UT  84108  NOT AVAILABLE  ...  2020/05/05 00:00:00   

                                                WEBSITE       STATE_ID  \
7594  https://aahs.org/locations/j--kent-mcnew-famil...  NOT AVAILABLE   
7595          https://healthcare.utah.edu/orthopaedics/  NOT AVAILABLE   

           ALT_NAME ST_FIPS               OWNER TTL_STAFF  BEDS  \
7594  NOT AVAILABLE      24          NON-PROFIT      -999  -999   
7595  NOT AVAILABLE      49  GOVERNMENT - STATE      -999  -999   

             TRAUMA  HELIPAD  
7594  NOT AVAILABLE        N  
7595  NOT AVAILABLE        N  

[2 rows x 34 columns]

Looking at the first and last few rows it looks like to be dataset of hsopitals from across the country with 
various details such as their location,website, number of beds etc. 

In [8]:
hospitals.shape

(7596, 34)

Number of rows is 7596 and columns is 34. 

In [9]:
hospitals.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7596 entries, 0 to 7595
Data columns (total 34 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   X           7596 non-null   float64
 1   Y           7596 non-null   float64
 2   OBJECTID    7596 non-null   int64  
 3   ID          7596 non-null   int64  
 4   NAME        7596 non-null   object 
 5   ADDRESS     7596 non-null   object 
 6   CITY        7596 non-null   object 
 7   STATE       7596 non-null   object 
 8   ZIP         7596 non-null   int64  
 9   ZIP4        7596 non-null   object 
 10  TELEPHONE   7596 non-null   object 
 11  TYPE        7596 non-null   object 
 12  STATUS      7596 non-null   object 
 13  POPULATION  7596 non-null   int64  
 14  COUNTY      7596 non-null   object 
 15  COUNTYFIPS  7596 non-null   object 
 16  COUNTRY     7596 non-null   object 
 17  LATITUDE    7596 non-null   float64
 18  LONGITUDE   7596 non-null   float64
 19  NAICS_CODE  7596 non-null  

In [10]:
hospitals_trimmed=hospitals[['NAME', 'ADDRESS', 'CITY', 'STATE', 'TYPE',  'POPULATION', 'COUNTY', 'COUNTYFIPS', 'NAICS_DESC', 'BEDS']]

In [12]:
hospitals_trimmed.shape

(7596, 10)

In [14]:
hospitals['TYPE'].value_counts()

GENERAL ACUTE CARE    4481
CRITICAL ACCESS       1027
PSYCHIATRIC            759
LONG TERM CARE         431
REHABILITATION         378
MILITARY               216
SPECIAL                145
CHILDREN               130
WOMEN                   20
CHRONIC DISEASE          9
Name: TYPE, dtype: int64

The different hospital types are 10 in number with General Acute Care being the most popular

hospitals_trimmed['CITY'].value_counts()

In [20]:
hospitals_trimmed_Nashville = hospitals_trimmed.loc[hospitals_trimmed['CITY'] == 'NASHVILLE']


In [23]:
hospitals_trimmed_Nashville_TN = hospitals_trimmed_Nashville.loc[hospitals_trimmed_Nashville['STATE'] == 'TN']

In [24]:
hospitals_trimmed_Nashville_TN.shape

(13, 10)

In [25]:
hospitals_trimmed_Nashville_TN

NAME  \
545                  TRISTAR SKYLINE MEDICAL CENTER   
3449          SELECT SPECIALTY HOSPITAL - NASHVILLE   
3485                           CURAHEALTH NASHVILLE   
5429             THE WOMEN'S HOSPITAL AT CENTENNIAL   
5433        METROPOLITAN NASHVILLE GENERAL HOSPITAL   
6289                  SAINT THOMAS MIDTOWN HOSPITAL   
6293  VANDERBILT STALLWORTH REHABILITATION HOSPITAL   
6661           VA MEDICAL CENTER - NASHVILLE CAMPUS   
6774    SAINT THOMAS HOSPITAL FOR SPECIALTY SURGERY   
6777                     SAINT THOMAS WEST HOSPITAL   
6786          TRISTAR SOUTHERN HILLS MEDICAL CENTER   
6808              TRISTAR CENTENNIAL MEDICAL CENTER   
6813                 VANDERBILT UNIVERSITY HOSPITAL   

                             ADDRESS       CITY STATE                TYPE  \
545              3441 DICKERSON PIKE  NASHVILLE    TN  GENERAL ACUTE CARE   
3449    2000 HAYES STREET, SUITE 500  NASHVILLE    TN      LONG TERM CARE   
3485  1412 COUNTY HOSPITAL ROAD, B-1  NASHVILLE    TN      LONG TERM CARE   
5429              2221 MURPHY AVENUE  NASHVILLE    TN               WOMEN   
5433              1818 ALBION STREET  NASHVILLE    TN  GENERAL ACUTE CARE   
6289              2000 CHURCH STREET  NASHVILLE    TN  GENERAL ACUTE CARE   
6293             2201 CHILDREN'S WAY  NASHVILLE    TN      REHABILITATION   
6661                 1310 24TH AVE S  NASHVILLE    TN            MILITARY   
6774              2011 MURPHY AVENUE  NASHVILLE    TN  GENERAL ACUTE CARE   
6777               4220 HARDING ROAD  NASHVILLE    TN  GENERAL ACUTE CARE   
6786                391 WALLACE ROAD  NASHVILLE    TN  GENERAL ACUTE CARE   
6808           2300 PATTERSON STREET  NASHVILLE    TN  GENERAL ACUTE CARE   
6813       1211 MEDICAL CENTER DRIVE  NASHVILLE    TN  GENERAL ACUTE CARE   

      POPULATION    COUNTY COUNTYFIPS  \
545          213  DAVIDSON      47037   
3449          70  DAVIDSON      47037   
3485          60  DAVIDSON      47037   
5429        -999  DAVIDSON      47037   
5433         150  DAVIDSON      47037   
6289         683  DAVIDSON      47037   
6293          80  DAVIDSON      47037   
6661         493  DAVIDSON      47037   
6774          23  DAVIDSON      47037   
6777         541  DAVIDSON      47037   
6786         126  DAVIDSON      47037   
6808         733  DAVIDSON      47037   
6813        1091  DAVIDSON      47037   

                                             NAICS_DESC  BEDS  
545              GENERAL MEDICAL AND SURGICAL HOSPITALS   213  
3449  SPECIALTY (EXCEPT PSYCHIATRIC AND SUBSTANCE AB...    70  
3485  SPECIALTY (EXCEPT PSYCHIATRIC AND SUBSTANCE AB...    60  
5429  SPECIALTY (EXCEPT PSYCHIATRIC AND SUBSTANCE AB...  -999  
5433             GENERAL MEDICAL AND SURGICAL HOSPITALS   150  
6289             GENERAL MEDICAL AND SURGICAL HOSPITALS   683  
6293  REHABILITATION HOSPITALS (EXCEPT ALCOHOLISM, D...    80  
6661             GENERAL MEDICAL AND SURGICAL HOSPITALS   493  
6774             GENERAL MEDICAL AND SURGICAL HOSPITALS    23  
6777             GENERAL MEDICAL AND SURGICAL HOSPITALS   541  
6786             GENERAL MEDICAL AND SURGICAL HOSPITALS   126  
6808             GENERAL MEDICAL AND SURGICAL HOSPITALS   733  
6813             GENERAL MEDICAL AND SURGICAL HOSPITALS  1091

The beds column is varied with some having as low as 23 and some having as high 1091. One hospital has -999 beds which 
did not made sense to me. 


In [33]:
hospitals_TN = hospitals.loc[hospitals['STATE'] == 'TN']

In [34]:
hospitals_TN

X             Y  OBJECTID         ID  \
515  -9.409026e+06  4.210878e+06       517   14737331   
535  -9.535209e+06  4.348723e+06       537    3438562   
536  -9.692242e+06  4.246701e+06       538    8038401   
537  -9.891187e+06  4.257288e+06       539    8338305   
540  -9.830658e+06  4.341502e+06       542    6538242   
...            ...           ...       ...        ...   
6810 -9.630429e+06  4.377577e+06      6823    2637148   
6811 -9.636394e+06  4.297187e+06      6824   10737167   
6812 -9.174179e+06  4.319023e+06      6825   10137650   
6813 -9.662773e+06  4.320018e+06      6826   15837232   
7362 -9.177696e+06  4.316931e+06      7376  195337650   

                                          NAME                    ADDRESS  \
515       STARR REGIONAL MEDICAL CENTER ETOWAH      886 HIGHWAY 411 NORTH   
535                    JACKSON COUNTY HOSPITAL         620 HOSPITAL DRIVE   
536                    MAURY REGIONAL HOSPITAL       1224 TROTWOOD AVENUE   
537   WEST TENNESSEE HEALTHCARE NORTH HOSPITAL         367 HOSPITAL BLVD.   
540                HENRY COUNTY MEDICAL CENTER           301 TYSON AVENUE   
...                                        ...                        ...   
6810             TRISTAR PORTLAND MEDICAL PARK             105 RED BUD DR   
6811         TRISTAR STONECREST MEDICAL CENTER   200 STONECREST BOULEVARD   
6812           UNICOI COUNTY MEMORIAL HOSPITAL        100 GREENWAY CIRCLE   
6813            VANDERBILT UNIVERSITY HOSPITAL  1211 MEDICAL CENTER DRIVE   
7362                    UNICOI COUNTY HOSPITAL      2030 TEMPLE HILL ROAD   

            CITY STATE    ZIP           ZIP4  ...             VAL_DATE  \
515       ETOWAH    TN  37331  NOT AVAILABLE  ...  2014/02/10 00:00:00   
535   GAINESBORO    TN  38562  NOT AVAILABLE  ...  2014/02/12 00:00:00   
536     COLUMBIA    TN  38401  NOT AVAILABLE  ...  2014/02/10 00:00:00   
537      JACKSON    TN  38305  NOT AVAILABLE  ...  2014/02/10 00:00:00   
540        PARIS    TN  38242  NOT AVAILABLE  ...  2014/02/10 00:00:00   
...          ...   ...    ...            ...  ...                  ...   
6810    PORTLAND    TN  37148  NOT AVAILABLE  ...  2014/02/12 00:00:00   
6811      SMYRNA    TN  37167  NOT AVAILABLE  ...  2014/02/10 00:00:00   
6812       ERWIN    TN  37650  NOT AVAILABLE  ...  2014/02/10 00:00:00   
6813   NASHVILLE    TN  37232  NOT AVAILABLE  ...  2014/02/10 00:00:00   
7362       ERWIN    TN  37650  NOT AVAILABLE  ...  2020/05/04 00:00:00   

                                                WEBSITE       STATE_ID  \
515                       http://www.starrregional.com/  NOT AVAILABLE   
535                                       NOT AVAILABLE         477446   
536                        http://www.mauryregional.com         490443   
537   http://www.tennovawest.com/locations/tennova-h...          76774   
540                              http://www.hcmc-tn.org         862759   
...                                                 ...            ...   
6810                      http://tristarportlander.com/         583205   
6811                      http://tristarstonecrest.com/         662440   
6812                             http://www.ucmhnet.org         687194   
6813                      http://www.mc.vanderbilt.edu/  NOT AVAILABLE   
7362  https://www.balladhealth.org/hospitals/unicoi-...  NOT AVAILABLE   

                                         ALT_NAME ST_FIPS  \
515                       WOODS MEMORIAL HOSPITAL      47   
535                                 NOT AVAILABLE      47   
536                                 NOT AVAILABLE      47   
537                                 NOT AVAILABLE      47   
540                                 NOT AVAILABLE      47   
...                                           ...     ...   
6810  TENNESSEE CHRISTIAN MEDICAL CENTER PORTLAND      47   
6811                                NOT AVAILABLE      47   
6812                                NOT AVAILABLE      47   
6813               

In [35]:
hospitals_TN['COUNTY'].nunique()


80

In [36]:
hospitals_TN['COUNTY'].value_counts()

SHELBY        23
DAVIDSON      15
HAMILTON      13
KNOX          11
WASHINGTON     6
              ..
JACKSON        1
MAURY          1
HARDEMAN       1
HARDIN         1
LOUDON         1
Name: COUNTY, Length: 80, dtype: int64

The Shelby county has the highest number of hospitals. 15 counties don't have a single hospital.